packages/static-module-record/src/transform-analyze.js

```diff
@@ -82,23 +82,25 @@ const makeCreateStaticRecord = transformSource =>
       })
       .join('');
 
-    // The functor captures the SES `arguments`, which is definitely
-    // less bad than the functor's arguments (which we are trying to
-    // hide).
-    //
-    // It must also be strict to enforce strictness of modules.
-    // We use destructuring parameters, so 'use strict' is not allowed
-    // but the function actually is strict.
+    // The outer function destructures the module calling convention's internal
+    // variables into hidden lexical variables, and also binds `this` to
+    // `undefined`.
+    // The inner function ensures that `arguments` is empty instead of
+    // revealing the module system internals or the evaluator's arguments.
+    // There is no avoiding the overshadowing of `globalThis.arguments` if it
+    // exists in this emulation of ESM since the evaluator binds `arguments` as
+    // well.
+    // Relies on the evaluator to ensure these functions are strict.
     let functorSource = `\
-(({ \
+(function ({ \
```

Comment:
For shedding, `this`, only the inner function matters. The outer function can remain an arrow function.

```suggestion
(({ \
```